pkg/common/cliprinter/structpretty/structpretty.go

```diff
@@ -0,0 +1,120 @@
+package structpretty
+
+import (
+	"fmt"
+	"io"
+	"reflect"
+	"strings"
+)
+
+// Print prints a struct prettily.
+// It will print only easily printable types, and only to one
+// level of depth. It will print arrays, slices, and maps if
+// their keys and elements are also easily printable types.
+func Print(msgs []interface{}, stdout, stderr io.Writer) bool {
+	if msgs == nil || len(msgs) == 0 {
+		return true
+	}
+
+	for _, msg := range msgs {
+		if msg == nil {
+			continue
+		}
+
+		printStruct(msg, stdout, stderr)
+	}
+
+	return true
+}
+
+func printStruct(msg interface{}, stdout, _ io.Writer) {
+	msgType := reflect.TypeOf(msg)
+	msgValue := reflect.ValueOf(msg)
+
+	// We also want to accept pointers to structs
+	if reflect.TypeOf(msg).Kind() == reflect.Ptr {
+		if reflect.TypeOf(msg).Elem().Kind() != reflect.Struct {
```

Comment:
Should this fail instead of silently succeed?

Reply:
Yes, fixed